In [1]:
# import libraries

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

plt.style.use('ggplot')
pd.set_option('display.max_columns', 200) 

In [2]:
df = pd.read_csv('AB_NYC_2019.csv', sep=',').copy()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [3]:
df.head().T

0  \
id                                                            2539   
name                            Clean & quiet apt home by the park   
host_id                                                       2787   
host_name                                                     John   
neighbourhood_group                                       Brooklyn   
neighbourhood                                           Kensington   
latitude                                                  40.64749   
longitude                                                -73.97237   
room_type                                             Private room   
price                                                          149   
minimum_nights                                                   1   
number_of_reviews                                                9   
last_review                                             2018-10-19   
reviews_per_month                                             0.21   
calculated_host_listings_count                                   6   
availability_365                                               365   

                                                    1  \
id                                               2595   
name                            Skylit Midtown Castle   
host_id                                          2845   
host_name                                    Jennifer   
neighbourhood_group                         Manhattan   
neighbourhood                                 Midtown   
latitude                                     40.75362   
longitude                                   -73.98377   
room_type                             Entire home/apt   
price                                             225   
minimum_nights                                      1   
number_of_reviews                                  45   
last_review                                2019-05-21   
reviews_per_month                                0.38   
calculated_host_listings_count                      2   
availability_365                                  355   

                                                                  2  \
id                                                             3647   
name                            THE VILLAGE OF HARLEM....NEW YORK !   
host_id                                                        4632   
host_name                                                 Elisabeth   
neighbourhood_group                                       Manhattan   
neighbourhood                                                Harlem   
latitude                                                   40.80902   
longitude                                                  -73.9419   
room_type                                              Private room   
price                                                           150   
minimum_nights                                                    3   
number_of_reviews                                                 0   
last_review                                                     NaN   
reviews_per_month                                               NaN   
calculated_host_listings_count                                    1   
availability_365                                                365   

                                                              3  \
id                                                         3831   
name                            Cozy Entire Floor of Brownstone   
host_id                                                    4869   
host_name                                           LisaRoxanne   
neighbourhood_group                                    Brooklyn   
neighbourhood                                      Clinton Hill   
latitude                                               40.68514   
longitude                                             -73.95976   
room_type                                       Entire home/apt   
price                     

In [4]:
# drop unnecessary columns

df.drop(columns='host_name', inplace=True)

In [5]:
# rename columns

df.rename(columns={
    'id': 'listing_id',
    'name': 'listing_name',
    'number_of_reviews': 'total_reviews',
    'last_review': 'last_review_date',
    'calculated_host_listings_count': 'host_total_listings',
    'availability_365': 'available_days_in_year',
    }, inplace=True)

df.tail()

listing_id                                       listing_name  \
48890    36484665    Charming one bedroom - newly renovated rowhouse   
48891    36485057      Affordable room in Bushwick/East Williamsburg   
48892    36485431            Sunny Studio at Historical Neighborhood   
48893    36485609               43rd St. Time Square-cozy single bed   
48894    36487245  Trendy duplex in the very heart of Hell's Kitchen   

        host_id neighbourhood_group       neighbourhood  latitude  longitude  \
48890   8232441            Brooklyn  Bedford-Stuyvesant  40.67853  -73.94995   
48891   6570630            Brooklyn            Bushwick  40.70184  -73.93317   
48892  23492952           Manhattan              Harlem  40.81475  -73.94867   
48893  30985759           Manhattan      Hell's Kitchen  40.75751  -73.99112   
48894  68119814           Manhattan      Hell's Kitchen  40.76404  -73.98933   

             room_type  price  minimum_nights  total_reviews last_review_date  \
48890     Private room     70               2              0              NaN   
48891     Private room     40               4              0              NaN   
48892  Entire home/apt    115              10              0              NaN   
48893      Shared room     55               1              0              NaN   
48894     Private room     90               7              0              NaN   

       reviews_per_month  host_total_listings  available_days_in_year  
48890                NaN                    2                       9  
48891                NaN                    2                      36  
48892                NaN                    1                      27  
48893                NaN                    6                       2  
48894                NaN                    1                      23

In [6]:
df = df[['listing_id', 'listing_name', 
         'available_days_in_year',
        'neighbourhood', 'neighbourhood_group', 
        'room_type', 'price','minimum_nights', 
        'total_reviews', 'last_review_date', 'reviews_per_month',
        'latitude', 'longitude',
        'host_id', 'host_total_listings',        
        ]]

df.head()

listing_id                                      listing_name  \
0        2539                Clean & quiet apt home by the park   
1        2595                             Skylit Midtown Castle   
2        3647               THE VILLAGE OF HARLEM....NEW YORK !   
3        3831                   Cozy Entire Floor of Brownstone   
4        5022  Entire Apt: Spacious Studio/Loft by central park   

   available_days_in_year neighbourhood neighbourhood_group        room_type  \
0                     365    Kensington            Brooklyn     Private room   
1                     355       Midtown           Manhattan  Entire home/apt   
2                     365        Harlem           Manhattan     Private room   
3                     194  Clinton Hill            Brooklyn  Entire home/apt   
4                       0   East Harlem           Manhattan  Entire home/apt   

   price  minimum_nights  total_reviews last_review_date  reviews_per_month  \
0    149               1              9       2018-10-19               0.21   
1    225               1             45       2019-05-21               0.38   
2    150               3              0              NaN                NaN   
3     89               1            270       2019-07-05               4.64   
4     80              10              9       2018-11-19               0.10   

   latitude  longitude  host_id  host_total_listings  
0  40.64749  -73.97237     2787                    6  
1  40.75362  -73.98377     2845                    2  
2  40.80902  -73.94190     4632                    1  
3  40.68514  -73.95976     4869                    1  
4  40.79851  -73.94399     7192                    1

* review columns, values and data types

In [7]:
df.describe(include='all').T

count unique              top   freq  \
listing_id              48895.0    NaN              NaN    NaN   
listing_name              48879  47905   Hillside Hotel     18   
available_days_in_year  48895.0    NaN              NaN    NaN   
neighbourhood             48895    221     Williamsburg   3920   
neighbourhood_group       48895      5        Manhattan  21661   
room_type                 48895      3  Entire home/apt  25409   
price                   48895.0    NaN              NaN    NaN   
minimum_nights          48895.0    NaN              NaN    NaN   
total_reviews           48895.0    NaN              NaN    NaN   
last_review_date          38843   1764       2019-06-23   1413   
reviews_per_month       38843.0    NaN              NaN    NaN   
latitude                48895.0    NaN              NaN    NaN   
longitude               48895.0    NaN              NaN    NaN   
host_id                 48895.0    NaN              NaN    NaN   
host_total_listings     48895.0    NaN              NaN    NaN   

                                  mean              std       min        25%  \
listing_id              19017143.23618   10983108.38561    2539.0  9471945.0   
listing_name                       NaN              NaN       NaN        NaN   
available_days_in_year      112.781327       131.622289       0.0        0.0   
neighbourhood                      NaN              NaN       NaN        NaN   
neighbourhood_group                NaN              NaN       NaN        NaN   
room_type                          NaN              NaN       NaN        NaN   
price                       152.720687        240.15417       0.0       69.0   
minimum_nights                7.029962         20.51055       1.0        1.0   
total_reviews                23.274466        44.550582       0.0        1.0   
last_review_date                   NaN              NaN       NaN        NaN   
reviews_per_month             1.373221         1.680442      0.01       0.19   
latitude                     40.728949          0.05453  40.49979    40.6901   
longitude                    -73.95217         0.046157 -74.24442  -73.98307   
host_id                 67620010.64661  78610967.032667    2438.0  7822033.0   
host_total_listings           7.143982        32.952519       1.0        1.0   

                               50%          75%          max  
listing_id              19677284.0   29152178.5   36487245.0  
listing_name                   NaN          NaN          NaN  
available_days_in_year        45.0        227.0        365.0  
neighbourhood                  NaN          NaN          NaN  
neighbourhood_group            NaN          NaN          NaN  
room_type                      NaN          NaN          NaN  
price                        106.0        175.0      10000.0  
minimum_nights                 3.0          5.0       1250.0  
total_reviews                  5.0         24.0        629.0  
last_review_date               NaN          NaN          NaN  
reviews_per_month             0.72         2.02         58.5  
latitude                  40.72307    40.763115     40.91306  
longitude                -73.95568   -73.936275    -73.71299  
host_id                 30793816.0  107434423.0  274321313.0  
host_total_listings            1.0          2.0        327.0

#### first insights:

* listing_id values are unique for each rows but unique listing_name are less
* average available days of a property is 45 days in a year
* minimum_nights has some extremely high values up to 1250
* host_total_listings max value is 327

In [8]:
df.duplicated().sum()

0

In [9]:
df.isna().sum()

listing_id                    0
listing_name                 16
available_days_in_year        0
neighbourhood                 0
neighbourhood_group           0
room_type                     0
price                         0
minimum_nights                0
total_reviews                 0
last_review_date          10052
reviews_per_month         10052
latitude                      0
longitude                     0
host_id                       0
host_total_listings           0
dtype: int64

In [10]:
# view nan values in listing_name column

df_filtered = df[df['listing_name'].isna()]
df_filtered.head(25).T

2854             3703               5775   \
listing_id                        1615764          2232600            4209595   
listing_name                          NaN              NaN                NaN   
available_days_in_year                362              341                  0   
neighbourhood           Battery Park City     East Village  Greenwich Village   
neighbourhood_group             Manhattan        Manhattan          Manhattan   
room_type                 Entire home/apt  Entire home/apt    Entire home/apt   
price                                 400              200                225   
minimum_nights                       1000                1                  1   
total_reviews                           0               28                  1   
last_review_date                      NaN       2015-06-08         2015-01-01   
reviews_per_month                     NaN             0.45               0.02   
latitude                         40.71239         40.73215           40.73473   
longitude                        -74.0162        -73.98821          -73.99244   
host_id                           6676776         11395220           20700823   
host_total_listings                     1                1                  1   

                                  5975          6269          6567   \
listing_id                      4370230       4581788       4756856   
listing_name                        NaN           NaN           NaN   
available_days_in_year                0             0             0   
neighbourhood                    Nolita  Williamsburg      Bushwick   
neighbourhood_group           Manhattan      Brooklyn      Brooklyn   
room_type               Entire home/apt  Private room  Private room   
price                               215           150            70   
minimum_nights                        7             1             1   
total_reviews                         5             0             0   
last_review_date             2016-01-02           NaN           NaN   
reviews_per_month                  0.09           NaN           NaN   
latitude                       40.72046       40.7137      40.70046   
longitude                      -73.9955     -73.94378     -73.92825   
host_id                        22686810      21600904       1832442   
host_total_listings                   1             1             1   

                                     6605          8841             11963  \
listing_id                         4774658       6782407          9325951   
listing_name                           NaN           NaN              NaN   
available_days_in_year                   0             0                0   
neighbourhood           Washington Heights  Williamsburg   Hell's Kitchen   
neighbourhood_group              Manhattan      Brooklyn        Manhattan   
room_type                     Private room  Private room  Entire home/apt   
price                                   40            45              190   
minimum_nights                           1             1                4   
total_reviews                            0             0                1   
last_review_date                       NaN           NaN       2016-01-05   
reviews_per_month                      NaN           NaN             0.02   
latitude                          40.85198      40.71354         40.76436   
longitude                        -73.93108     -73.93882        -73.98573   
host_id                           24625694      31147528         33377685   
host_total_listings                      1             1                1   

                                  12824         13059         13401  \
listing_id                      9787590       9885866      10052289   
listing_name                        NaN           NaN           NaN   
available_days_in_year                0             0           362   
neighbourhood                    Harlem     Chinatown   Brownsville   
neighbourhood_

* they represent real rooms only with missing names, these may not have been booked during the year
* for these rows, I will assign a default name such as 'Untitled Listing'

In [11]:
# replace missing listing names with 'Untitled Listing'

df['listing_name'].fillna('Untitled Listing', inplace=True)

In [12]:
# view 0 values in available_days_in_year column

df_filtered = df[df['available_days_in_year'] == 0]
df_filtered.head(25).T

4    \
listing_id                                                          5022   
listing_name            Entire Apt: Spacious Studio/Loft by central park   
available_days_in_year                                                 0   
neighbourhood                                                East Harlem   
neighbourhood_group                                            Manhattan   
room_type                                                Entire home/apt   
price                                                                 80   
minimum_nights                                                        10   
total_reviews                                                          9   
last_review_date                                              2018-11-19   
reviews_per_month                                                    0.1   
latitude                                                        40.79851   
longitude                                                      -73.94399   
host_id                                                             7192   
host_total_listings                                                    1   

                                       6    \
listing_id                            5121   
listing_name               BlissArtsSpace!   
available_days_in_year                   0   
neighbourhood           Bedford-Stuyvesant   
neighbourhood_group               Brooklyn   
room_type                     Private room   
price                                   60   
minimum_nights                          45   
total_reviews                           49   
last_review_date                2017-10-05   
reviews_per_month                      0.4   
latitude                          40.68688   
longitude                        -73.95596   
host_id                               7356   
host_total_listings                      1   

                                                       8    \
listing_id                                            5203   
listing_name            Cozy Clean Guest Room - Family Apt   
available_days_in_year                                   0   
neighbourhood                              Upper West Side   
neighbourhood_group                              Manhattan   
room_type                                     Private room   
price                                                   79   
minimum_nights                                           2   
total_reviews                                          118   
last_review_date                                2017-07-21   
reviews_per_month                                     0.99   
latitude                                          40.80178   
longitude                                        -73.96723   
host_id                                               7490   
host_total_listings                                      1   

                                                  14   \
listing_id                                       6090   
listing_name            West Village Nest - Superhost   
available_days_in_year                              0   
neighbourhood                            West Village   
neighbourhood_group                         Manhattan   
room_type                             Entire home/apt   
price                                             120   
minimum_nights                                     90   
total_reviews                                      27   
last_review_date                           2018-10-31   
reviews_per_month                                0.22   
latitude                                      40.7353   
longitude                                   -74.00525   
host_id                                         11975   
host_total_listings                                 1   

                                                     20   \
listing_id                                          7801   
listing_name            Sweet and Spacious Brooklyn Loft   
available_days_in_year       

* even they may represent valid records that were simply unavailable during the last year, they should be removed to avoid calculation issues and skewed results 

In [13]:
df = df[df['available_days_in_year'] != 0].reset_index(drop=True)

In [14]:
# view nan values in last_review_date column

df_filtered = df[df['last_review_date'].isna()]
df_filtered.head(25).T

2    \
listing_id                                             3647   
listing_name            THE VILLAGE OF HARLEM....NEW YORK !   
available_days_in_year                                  365   
neighbourhood                                        Harlem   
neighbourhood_group                               Manhattan   
room_type                                      Private room   
price                                                   150   
minimum_nights                                            3   
total_reviews                                             0   
last_review_date                                        NaN   
reviews_per_month                                       NaN   
latitude                                           40.80902   
longitude                                          -73.9419   
host_id                                                4632   
host_total_listings                                       1   

                                                      15   \
listing_id                                           7750   
listing_name            Huge 2 BR Upper East  Cental Park   
available_days_in_year                                249   
neighbourhood                                 East Harlem   
neighbourhood_group                             Manhattan   
room_type                                 Entire home/apt   
price                                                 190   
minimum_nights                                          7   
total_reviews                                           0   
last_review_date                                      NaN   
reviews_per_month                                     NaN   
latitude                                         40.79685   
longitude                                       -73.94872   
host_id                                             17985   
host_total_listings                                     2   

                                                30   \
listing_id                                    11452   
listing_name            Clean and Quiet in Brooklyn   
available_days_in_year                          365   
neighbourhood                    Bedford-Stuyvesant   
neighbourhood_group                        Brooklyn   
room_type                              Private room   
price                                            35   
minimum_nights                                   60   
total_reviews                                     0   
last_review_date                                NaN   
reviews_per_month                               NaN   
latitude                                   40.68876   
longitude                                 -73.94312   
host_id                                        7355   
host_total_listings                               1   

                                              32   \
listing_id                                  11943   
listing_name            Country space in the city   
available_days_in_year                        365   
neighbourhood                            Flatbush   
neighbourhood_group                      Brooklyn   
room_type                            Private room   
price                                         150   
minimum_nights                                  1   
total_reviews                                   0   
last_review_date                              NaN   
reviews_per_month                             NaN   
latitude                                 40.63702   
longitude                               -73.96327   
host_id                                     45445   
host_total_listings                             1   

                                                        182           230  \
listing_id                                            54466         63588   
listing_name            Beautiful Uptown Manhattan apartmnt           LL3   
available_days_in_year                                  365           349   
neighbourhood                   

In [15]:
# check if the missing values in both columns are in the same rows

if df['last_review_date'].isna().equals(df['reviews_per_month'].isna()):
    print("yes they are same.")
else:
    print("nope, check it")


yes they are same.


In [16]:
df_filtered.info()

<class 'pandas.core.frame.DataFrame'>
Index: 5207 entries, 2 to 31361
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   listing_id              5207 non-null   int64  
 1   listing_name            5207 non-null   object 
 2   available_days_in_year  5207 non-null   int64  
 3   neighbourhood           5207 non-null   object 
 4   neighbourhood_group     5207 non-null   object 
 5   room_type               5207 non-null   object 
 6   price                   5207 non-null   int64  
 7   minimum_nights          5207 non-null   int64  
 8   total_reviews           5207 non-null   int64  
 9   last_review_date        0 non-null      object 
 10  reviews_per_month       0 non-null      float64
 11  latitude                5207 non-null   float64
 12  longitude               5207 non-null   float64
 13  host_id                 5207 non-null   int64  
 14  host_total_listings     5207 non-null   int6

* these records may represent new or low-traffic rooms that did not receive any reviews or bookings but they may also extremely skew results and calculations so i am removing these as well  

In [17]:
df = df.dropna(subset=['last_review_date']).reset_index(drop=True)
df.head()

listing_id                               listing_name  \
0        2539         Clean & quiet apt home by the park   
1        2595                      Skylit Midtown Castle   
2        3831            Cozy Entire Floor of Brownstone   
3        5099  Large Cozy 1 BR Apartment In Midtown East   
4        5178           Large Furnished Room Near B'way    

   available_days_in_year   neighbourhood neighbourhood_group  \
0                     365      Kensington            Brooklyn   
1                     355         Midtown           Manhattan   
2                     194    Clinton Hill            Brooklyn   
3                     129     Murray Hill           Manhattan   
4                     220  Hell's Kitchen           Manhattan   

         room_type  price  minimum_nights  total_reviews last_review_date  \
0     Private room    149               1              9       2018-10-19   
1  Entire home/apt    225               1             45       2019-05-21   
2  Entire home/apt     89               1            270       2019-07-05   
3  Entire home/apt    200               3             74       2019-06-22   
4     Private room     79               2            430       2019-06-24   

   reviews_per_month  latitude  longitude  host_id  host_total_listings  
0               0.21  40.64749  -73.97237     2787                    6  
1               0.38  40.75362  -73.98377     2845                    2  
2               4.64  40.68514  -73.95976     4869                    1  
3               0.59  40.74767  -73.97500     7322                    1  
4               3.47  40.76489  -73.98493     8967                    1

In [18]:
df.dtypes

listing_id                  int64
listing_name               object
available_days_in_year      int64
neighbourhood              object
neighbourhood_group        object
room_type                  object
price                       int64
minimum_nights              int64
total_reviews               int64
last_review_date           object
reviews_per_month         float64
latitude                  float64
longitude                 float64
host_id                     int64
host_total_listings         int64
dtype: object

In [19]:
df.nunique()

listing_id                26155
listing_name              25879
available_days_in_year      365
neighbourhood               217
neighbourhood_group           5
room_type                     3
price                       564
minimum_nights               84
total_reviews               387
last_review_date           1216
reviews_per_month           929
latitude                  14652
longitude                 12316
host_id                   18944
host_total_listings          47
dtype: int64

In [20]:
# convert 'neighbourhood' and 'room_type' to category

df['neighbourhood'] = df['neighbourhood'].astype('category')
df['room_type'] = df['room_type'].astype('category')

In [21]:
df.isna().sum()

listing_id                0
listing_name              0
available_days_in_year    0
neighbourhood             0
neighbourhood_group       0
room_type                 0
price                     0
minimum_nights            0
total_reviews             0
last_review_date          0
reviews_per_month         0
latitude                  0
longitude                 0
host_id                   0
host_total_listings       0
dtype: int64

In [22]:
df.describe(include='all').T

count unique                               top  \
listing_id              26155.0    NaN                               NaN   
listing_name              26155  25879  Loft Suite @ The Box House Hotel   
available_days_in_year  26155.0    NaN                               NaN   
neighbourhood             26155    217                Bedford-Stuyvesant   
neighbourhood_group       26155      5                         Manhattan   
room_type                 26155      3                   Entire home/apt   
price                   26155.0    NaN                               NaN   
minimum_nights          26155.0    NaN                               NaN   
total_reviews           26155.0    NaN                               NaN   
last_review_date          26155   1216                        2019-06-23   
reviews_per_month       26155.0    NaN                               NaN   
latitude                26155.0    NaN                               NaN   
longitude               26155.0    NaN                               NaN   
host_id                 26155.0    NaN                               NaN   
host_total_listings     26155.0    NaN                               NaN   

                         freq             mean              std       min  \
listing_id                NaN  19320255.560122  11210938.636519    2539.0   
listing_name               11              NaN              NaN       NaN   
available_days_in_year    NaN       170.613496       124.153024       1.0   
neighbourhood            2212              NaN              NaN       NaN   
neighbourhood_group     10813              NaN              NaN       NaN   
room_type               13709              NaN              NaN       NaN   
price                     NaN       149.885184       198.816962       0.0   
minimum_nights            NaN         6.585739        19.602095       1.0   
total_reviews             NaN        38.193615        54.359918       1.0   
last_review_date         1323              NaN              NaN       NaN   
reviews_per_month         NaN         1.810126         1.805707      0.01   
latitude                  NaN        40.727482         0.056936  40.50641   
longitude                 NaN       -73.947724         0.051476 -74.24442   
host_id                   NaN  74574778.668591  82617013.602094    2571.0   
host_total_listings       NaN         6.989715        31.807739       1.0   

                              25%         50%          75%          max  
listing_id              9457277.5  20783233.0   29509166.5   36455809.0  
listing_name                  NaN         NaN          NaN          NaN  
available_days_in_year       53.0       160.0        292.0        365.0  
neighbourhood                 NaN         NaN          NaN          NaN  
neighbourhood_group           NaN         NaN          NaN          NaN  
room_type                     NaN         NaN          NaN          NaN  
price                        70.0       109.0        175.0       9999.0  
minimum_nights                1.0         2.0          4.0       1250.0  
total_reviews                 5.0        16.0         49.0        629.0  
last_review_date              NaN         NaN          NaN          NaN  
reviews_per_month            0.45        1.23         2.68         58.5  
latitude                 40.68703    40.72167     40.76303     40.91306  
longitude               -73.98241   -73.95304    -73.92922    -73.71299  
host_id                 7365834.0  35281088.0  131647128.0  273841667.0  
host_total_listings           1.0         1.0          3.0        327.0

In [23]:
# review price column with 0 values 

df_filtered = df[df['price'] == 0]
df_filtered.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8 entries, 11660 to 13533
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype   
---  ------                  --------------  -----   
 0   listing_id              8 non-null      int64   
 1   listing_name            8 non-null      object  
 2   available_days_in_year  8 non-null      int64   
 3   neighbourhood           8 non-null      category
 4   neighbourhood_group     8 non-null      object  
 5   room_type               8 non-null      category
 6   price                   8 non-null      int64   
 7   minimum_nights          8 non-null      int64   
 8   total_reviews           8 non-null      int64   
 9   last_review_date        8 non-null      object  
 10  reviews_per_month       8 non-null      float64 
 11  latitude                8 non-null      float64 
 12  longitude               8 non-null      float64 
 13  host_id                 8 non-null      int64   
 14  host_total_listings     8 n

In [24]:
df_filtered.head(10).T

11660  \
listing_id                                                       18750597   
listing_name            Huge Brooklyn Brownstone Living, Close to it all.   
available_days_in_year                                                 28   
neighbourhood                                          Bedford-Stuyvesant   
neighbourhood_group                                              Brooklyn   
room_type                                                    Private room   
price                                                                   0   
minimum_nights                                                          4   
total_reviews                                                           1   
last_review_date                                               2018-01-06   
reviews_per_month                                                    0.05   
latitude                                                         40.69023   
longitude                                                       -73.95428   
host_id                                                           8993084   
host_total_listings                                                     4   

                                                                12769  \
listing_id                                                   20333471   
listing_name            ★Hostel Style Room | Ideal Traveling Buddies★   
available_days_in_year                                            127   
neighbourhood                                         East Morrisania   
neighbourhood_group                                             Bronx   
room_type                                                Private room   
price                                                               0   
minimum_nights                                                      2   
total_reviews                                                      55   
last_review_date                                           2019-06-24   
reviews_per_month                                                2.56   
latitude                                                     40.83296   
longitude                                                   -73.88668   
host_id                                                     131697576   
host_total_listings                                                 4   

                                                                12937  \
listing_id                                                   20624541   
listing_name            Modern apartment in the heart of Williamsburg   
available_days_in_year                                             73   
neighbourhood                                            Williamsburg   
neighbourhood_group                                          Brooklyn   
room_type                                             Entire home/apt   
price                                                               0   
minimum_nights                                                      5   
total_reviews                                                       3   
last_review_date                                           2018-01-02   
reviews_per_month                                                0.15   
latitude                                                     40.70838   
longitude                                                   -73.94645   
host_id                                                      10132166   
host_total_listings                                                 1   

                                                                    12946  \
listing_id                                                       20639628   
listing_name            Spacious comfortable master bedroom with nice ...   
available_days_in_year                                                176   
neighbourhood                                          Bedford-Stuyvesant   
neighbourhood_group                                              Brooklyn   
room_type                          

In [25]:
df = df[df['price'] != 0].reset_index(drop=True)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26147 entries, 0 to 26146
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype   
---  ------                  --------------  -----   
 0   listing_id              26147 non-null  int64   
 1   listing_name            26147 non-null  object  
 2   available_days_in_year  26147 non-null  int64   
 3   neighbourhood           26147 non-null  category
 4   neighbourhood_group     26147 non-null  object  
 5   room_type               26147 non-null  category
 6   price                   26147 non-null  int64   
 7   minimum_nights          26147 non-null  int64   
 8   total_reviews           26147 non-null  int64   
 9   last_review_date        26147 non-null  object  
 10  reviews_per_month       26147 non-null  float64 
 11  latitude                26147 non-null  float64 
 12  longitude               26147 non-null  float64 
 13  host_id                 26147 non-null  int64   
 14  host_total_listings   

In [26]:
df['avg_price_per_night'] = df['price'] / df['minimum_nights']

In [27]:
df['avg_price_per_night'].describe()

count    26147.000000
mean        68.909108
std        127.430973
min          0.079079
25%         22.500000
50%         46.500000
75%         83.333333
max       8000.000000
Name: avg_price_per_night, dtype: float64

In [28]:
df[df['avg_price_per_night'] < 10]

listing_id                                       listing_name  \
35          12627  Entire apartment in central Brooklyn neighborh...   
44          14377                   Williamsburg 1 bedroom Apartment   
48          15341      **Bright Nolita Apt w Doorman/Elevators/Gym**   
50          15396                 Sunny & Spacious Chelsea Apartment   
52          16326         Comfortable 4-bedroom apt in family house.   
...           ...                                                ...   
26035    36032998        Spacious Sunny Private Bedroom with Windows   
26039    36038287             Huge, Clean & Quiet room in a 2 BR Apt   
26086    36137740                            Cozy Kips Bay Apartment   
26129    36280646                       Cable and wfi, L/G included.   
26138    36351543              2-MONTH SUBLEASE (WITH EARLY MOVE-IN)   

       available_days_in_year              neighbourhood neighbourhood_group  \
35                         95  Prospect-Lefferts Gardens            Brooklyn   
44                         30               Williamsburg            Brooklyn   
48                        154                       SoHo           Manhattan   
50                        180                    Chelsea           Manhattan   
52                        297           Prospect Heights            Brooklyn   
...                       ...                        ...                 ...   
26035                      40               Williamsburg            Brooklyn   
26039                     321            Stuyvesant Town           Manhattan   
26086                      54                   Kips Bay           Manhattan   
26129                     322               Forest Hills              Queens   
26138                      87         Bedford-Stuyvesant            Brooklyn   

             room_type  price  minimum_nights  total_reviews last_review_date  \
35     Entire home/apt    150              29             11       2019-06-05   
44     Entire home/apt    150              30            105       2019-06-22   
48     Entire home/apt    180              30             21       2019-06-29   
50     Entire home/apt    375             180              5       2018-11-03   
52     Entire home/apt    200              30            143       2019-01-26   
...                ...    ...             ...            ...              ...   
26035     Private room     44               5              1       2019-07-08   
26039     Private room     77              21              2       2019-06-25   
26086  Entire home/apt    130              30              1       2019-06-28   
26129  Entire home/apt     16               9              1       2019-07-07   
26138     Private room     33              30              2       2019-07-06   

       reviews_per_month  latitude  longitude    host_id  host_total_listings  \
35                  0.49  40.65944  -73.96238      49670                    1   
44                  0.90  40.70881  -73.95930      56512                    1   
48                  0.30  40.72210  -73.99775      60049                    1   
50                  0.12  40.74623  -73.99530      60278                    1   
52                  1.33  40.67811  -73.96428      63588                    2   
...                  ...       ...        ...        ...                  ...   
26035               1.00  40.71808  -73.94261    2355573                    2   
26039               2.00  40.73221  -73.97941  167769983                    1   
26086               1.00  40.74133  -73.97706   30933227                    1   
26129               1.00  40.73657  -73.85088  272872092                    1   
26138               2.00  40.68914  -73.92408   52984497                    1   

       avg_price_per_night  
35                5.172414  
44                5.000000  
48                6.000000  
50                2.083333  
52                6.666667  
...                    ...  
26035             8.800000  
260

In [29]:
# save data to use for Tableau dashboard

df.to_csv('AB_NYC_2019_clean.csv', index=False)

print("file saved")

file saved
